runtime/rxjava2-extensions/src/main/kotlin/com/squareup/sqldelight/runtime/rx/RxJavaExtensions.kt

```diff
@@ -0,0 +1,82 @@
+@file:JvmName("RxQuery")
+
+package com.squareup.sqldelight.runtime.rx
+
+import com.squareup.sqldelight.Query
+import io.reactivex.Observable
+import io.reactivex.ObservableEmitter
+import io.reactivex.ObservableOnSubscribe
+import io.reactivex.Scheduler
+import io.reactivex.annotations.CheckReturnValue
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import java.util.Optional
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Turns this [Query] into an [Observable] which emits whenever the underlying result set changes.
+ *
+ * ### Scheduler:
+ *   [asObservable] by default operates on the [io.reactivex.schedulers.Schedulers.io] scheduler but
+ *   can be optionally overridden with [scheduler]
+ */
+@CheckReturnValue
+@JvmOverloads
+@JvmName("toObservable")
+fun <T : Any> Query<T>.asObservable(scheduler: Scheduler = Schedulers.io()): Observable<Query<T>> {
```

Comment:
I'm pulling `asWhatever` from the Kotlin stdlib which is used when wrapping an instance into another. `toWhatever` is used when copying data from one to another, but it seems more appropriate for the Java version since the call-site syntax is different.